tensorflow/core/kernels/resize_bilinear_op_gpu.cu.cc

```diff
@@ -228,6 +229,56 @@ __global__ void ResizeBilinearGradKernel(const int32 nthreads,
   }
 }
 
+template <typename T>
+__global__ void ResizeBilinearDeterministicGradKernel(
+    const int32 nthreads, const float* __restrict__ input_grad,
+    float height_scale, float inverse_height_scale, float width_scale,
+    float inverse_width_scale, int batch, int original_height,
+    int original_width, int channels, int resized_height, int resized_width,
+    T* __restrict__ output_grad) {
+  GPU_1D_KERNEL_LOOP(out_idx, nthreads) {
+    // out_idx = c + channels * (x + original_width * (y + original_height * b))
+    int idx = out_idx;
+    const int c = idx % channels;
+    idx /= channels;
+    const int out_x_center = idx % original_width;
+    idx /= original_width;
+    const int out_y_center = idx % original_height;
+    const int b = idx / original_height;
+
+    int in_y_start = max(0, __float2int_ru(
+        (out_y_center - 1 + 0.5) * inverse_height_scale - 0.5));
```

Comment:
It looks like this would promote to double?

Reply:
Good point. The whole argument to `__float2int_ru()` will end up being double. This issue happens on the next line of code as well, and then of course for the x dimension as well. I'm considering changing every instance of `0.5` to `0.5f`.

Reply:
The most recent commit replaces these magic numbers (`0.5`) with the `offset` variable (of type `float`).